src/main/java/uk/gov/companieshouse/model/Reason.java

```diff
@@ -26,23 +28,41 @@ public void setOther(OtherReason other) {
         this.other = other;
     }
 
+    public IllnessReason getIllnessReason() { return this.illnessReason; }
+
+    public void setIllnessReason(IllnessReason illnessReason) {this.illnessReason = illnessReason; }
+
+    public ReasonType getReasonType() {
+        if(getOther() == null ^ getIllnessReason() == null){
+            return getOther() == null? getIllnessReason():getOther();
+        }
+        else {
+            return null;
+        }
+    }
+
     @Override
-    public boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
-        Reason reason = (Reason) o;
-        return Objects.equals(other, reason.other);
+    public boolean equals(final Object o) {
```

Comment:
Do we need this method? And hashCode() and toString()? Its looking unused. We can improve test coverage if we delete it :)

Reply:
These are standard method definitions but i suppose they can be removed

Reply:
Could we also do them consistently? In some places you are using an Apache library EqualsBuilder and HashBuilder to make them. I would prefer to see them done automatically by java unless we have a specific reason to customise the value.

Reply:
They have been removed. For consistency we will follow the practice already implemented. Thanks again Jonathan